contrib/k8s/README.md

```diff
@@ -1,41 +1,123 @@
-# dex
+# Deploying dex on Kubernetes
 
-## Getting Started
+This document will allow you to set up dex in your Kubernetes; the configuration files used are generally useful, but will need to be modified to meet the needs of your deployment. The places that are likely to need modification will be called out as often as possible in this document.
 
-**Warning**: Hacks Ahead.
+## Prerequisites and Assumptions
 
-You must be running cluster wide DNS for this to work.
+The document assumes that you already have a cluster with at least one worker up and running. The easiest way to bring up a small cluster for experimentation is the [coreos-kubernetes single node](coreos-kubernetes-single-node) Vagrant installer.
 
-Install your dockercfg. There is no nice way to do this:
+The other assumption is that your Kubernetes cluster will be routable on `172.17.4.99`  (which is what it will be if you use [coreos-kubernetes single node][coreos-kubernetes-single-node], and the issuer URL for your dex installation is `http://172.17.4.99:30556`; in production installations you will need to make sure that you are serving on https and you will likely want a host name on the normal TLS port instead of an IP address.
 
-```
-ssh worker
-cat > /proc/$(pgrep kubelet)/cwd/.dockercfg
-```
+[coreos-kubernetes-single-node][https://github.com/coreos/coreos-kubernetes/blob/master/single-node/README.md]
+
+## Start Postgres
 
-Start postgres
+Dex needs a database to store information; these commands will create a Postgres service that dex can use. Note that this configuration is not suitable for production - if the container is destroyed, the data is gone forever. 
+
+In production you should have a sufficiently fault-tolerant Postgres deployment on a persistent volume with backup.
 
 ```
 kubectl create -f postgres-rc.yaml
 kubectl create -f postgres-service.yaml
 ```
 
-Run dex and setup services
+## Create your secrets.
+
+dex needs a secret key for encrypting private keys in the database. These can be stored as [Kubernetes secrets][k8s-secrets].
+
+[k8s-secrets]: http://kubernetes.io/v1.0/docs/user-guide/secrets.html
+
+```
+kubectl create -f dex-secrets.yaml
+```
+
+## Start the Overlord
+
+Start the overlord. This will also initialize your database the first time it's run, and perform migrations when new versions are installed.
+
+```
+kubectl create -f dex-overlord-rc.yaml
+kubectl create -f dex-overlord-service.yaml
+```
+
+Note: this will expose the admin API on the cluster, which is essentially all-powerful. This will be fixed very soon. 
```

Comment:
Nit: Be more specific than "all-powerful". What does this mean "on the cluster" practically mean / what is being fixed?